src/backends/gpu/stack.hpp

```diff
@@ -22,91 +25,85 @@ namespace gpu {
 template <typename T>
 class stack {
     using value_type = T;
-    using allocator = memory::managed_allocator<value_type>;
-
-    // The number of items of type value_type that can be stored in the stack
-    unsigned capacity_;
-
-    // The number of items that have been stored
-    unsigned size_;
+    template <typename U>
+    using allocator = memory::managed_allocator<U>;
 
-    // Memory containing the value buffer
-    // Stored in managed memory to facilitate host-side access of values
-    // pushed from kernels on the device.
-    value_type* data_;
+    using base_type = stack_base<value_type>;
```

Comment:
Isn't stack_base really stack_storage?

Reply:
True. You and @halfflat both noticed this. I went for the name `stack_storage`, and added a comment to explain this.